doc/HOWTO/documenting-macros.md

```diff
@@ -0,0 +1,53 @@
+Documenting Macros
+==================
+
+"make doc-nits" (or the check-docs Action) might tell you something like:
+
+~~~~text
+include/openssl/bio.h: macro BIO_set_dgram_origin(3) undocumented
+include/openssl/bio.h: macro BIO_get_dgram_origin(3) undocumented
+include/openssl/bio.h: macro BIO_set_dgram_dest(3) undocumented
+include/openssl/bio.h: macro BIO_get_dgram_dest(3) undocumented
+~~~~
+
+and you'll want to document this.
+
+So, create a new .pod file in doc/man3/FUNCTION.pod.
+
+You might be documenting several related functions in that file,
+but don't try to think about giving it a neutral name.
+Just use the name of the first function as the file name,
+so in this case I called it:
+
+~~~~text
+doc/man3/BIO_set_dgram_origin.pod
+~~~~
+
+if you do use an unrelated name (like BIO_dgram.pod) then you'll get
+a warning about that.
+
+Then, you'll notice that your new file is not in test/build.info.
+That file is created by running:
+
+~~~~text
+make generate_doc_buildinfo
+~~~~
+
+this will update doc/build.info.
+You should git add the result as `generate_doc_buildinfo` is not run on every build.
+
+With these two changes, running `make doc-nits` locally should
+now agree with you that you have documented all your new defines,
+but it might then complain:
+
+````text
+BIO_get_dgram_dest(3) is supposedly internal
+(maybe missing from other.syms) but is documented as public
+````
```

Comment:
Four backticks should only be used to fence a text which displays triple backticks, like this:
````
```
BIO_get_dgram_dest(3) is supposedly internal
(maybe missing from other.syms) but is documented as public
```
````

Reply:
With tilde, the language is required by the nit-checker, but I've changed it all to three backtick.

Reply:
make doc-nits on my laptop did not complain about language missing, but the CI did. Sigh.

Reply:
> > With tilde, the language is required by the nit-checker, but I've changed it all to three backtick.
> 
> Curious... I don't think that any of the available specs give reason for that distinction.

The requirement for adding a „text“ specifier is complete ridiculous imo and I heavily dislike it.  It has no benefit (syntax highlighting), it just violates the principle that the markup should be least disruptive to the reader of the raw text document. We have tons of markdown documentation which use four space indentation (my preferred method, because it’s not disruptive at all) or just triple backquotes (second best), and no nit checker has ever complained.

Was this check added recently? If yes, it ought to be removed.

Reply:
We did not allow fenced code blocks before. I'd agree that requiring the specifier is nonsense, could @mcr just add the MD040 as exclude_rule in https://github.com/openssl/openssl/blob/master/util/markdownlint.rb